forum/profiles/serializers.py

```diff
@@ -83,3 +83,35 @@ class Meta:
             'city',
             'description',
         ]
+
+
+class ViewedStratupReadSerializer(serializers.ModelSerializer):
+    
+    startup_id= serializers.StringRelatedField(source='startup.id', read_only=True)
+    company_name= serializers.StringRelatedField(source='startup.company_name', read_only=True)
+
+    
+    class Meta:
+        model = ViewedStratups
+        fields = [
+            'startup_id',
+            'company_name',
+            'viewed_at'
+        ]
+
+
+class ViewedStratupCreateSerializer(serializers.ModelSerializer):
+    
+    investor = serializers.PrimaryKeyRelatedField(
+        queryset=InvestorProfile.objects.all()
+    )
+    startup = serializers.PrimaryKeyRelatedField(
+        queryset=StartupProfile.objects.all()
+    )
+
+    class Meta:
+        model = ViewedStratups
+        fields = ['investor', 'startup', 'viewed_at']
+
+    def create(self, validated_data):
+        return ViewedStratups.objects.create(**validated_data)
```

Comment:
While the database constraint will handle uniqueness, it's better to provide user-friendly error messages by adding validation in the `CreateSerializer`.
_Example:_
```
def validate(self, data):
    if ViewedStratups.objects.filter(
        investor=data['investor'], 
        startup=data['startup'], 
        viewed_at__date=data['viewed_at'].date()
    ).exists():
        raise serializers.ValidationError("This startup has already been viewed by this investor today.")
    return data
```